starter-code/routes/auth-routes.js

```diff
@@ -0,0 +1,88 @@
+const bcrypt = require('bcrypt');
+const saltRounds = 10;
+const express = require('express');
+const authRoutes = express.Router();
+const User = require('../models/user');
+
+
+authRoutes.get('/', (req, res, err) => res.render('index'))
+
+authRoutes.get("/signup", (req, res, next) => {
+  res.render("auth/signup");
+});
+
+authRoutes.post('/signup', (req, res, next) => {
+  let username = req.body.username;
+  let password = req.body.password;
+  let salt = bcrypt.genSaltSync(saltRounds);
+  let hashPass = bcrypt.hashSync(password, salt);
```

Comment:
Maybe you want to hash stuff only when you *know* you'll need to so.

Reply:
That's indeed smarter 😁